tests/functional/test_credentials.py

```diff
@@ -472,6 +478,37 @@ def test_process_source_profile(self):
         }
         self.assertEqual(kwargs, expected_kwargs)
 
+    def test_web_identity_source_profile(self):
```

Comment:
It's probably worth adding a functional test to make sure we do not use the env vars when chaining.